drape/gl_functions.cpp

```diff
@@ -309,21 +280,25 @@ void GLFunctions::Init(dp::ApiVersion apiVersion)
     glMapBufferRangeFn = &::glMapBufferRangeEXT;
     glFlushMappedBufferRangeFn = &::glFlushMappedBufferRangeEXT;
   }
+#endif
+#if defined(OMIM_OS_MAC) || defined(OMIM_OS_ANDROID) || defined(OMIM_OS_LINUX) || defined(OMIM_OS_MOBILE)
   else if (CurrentApiVersion == dp::ApiVersion::OpenGLES3)
```

Comment:
else if inside #if block looks confusing for me .. Looking at the file, we can reformat it like this:

```
/// VAO
#if !defined(OMIM_OS_WINDOWS)
if (CurrentApiVersion == dp::ApiVersion::OpenGLES2)
{
#if defined(OMIM_OS_MAC)
...
#elif defined(OMIM_OS_LINUX)
...
#elif defined(OMIM_OS_ANDROID)
...
#elif defined(OMIM_OS_MOBILE)
...
#endif
}
else if (CurrentApiVersion == dp::ApiVersion::OpenGLES3)
{
  // OpenGL ES3 api is the same for all systems, except WIN.
  ...
}
else
{
  CHECK(false, ("Unknown Graphics API:", CurrentApiVersion));
}
#else  // OMIM_OS_WINDOWS
...
#endif
```

Reply:
That was exactly my first naive approach. And then I thought it would be an improvement over that, to not nest the macro conditionals into each other, as that would increase the McCabe Cyclomatic complexity of the macro code, which is usually not desired and tends towards the so called "macro hell".
Now if you say that two levels of nesting is till perfectly fine and/or better than starting a block with a runtime `else if`, then I can revert back in no time. Should I?

Reply:
Wow, "McCabe Cyclomatic complexity" - I hear it for the first time for more than 20 years of programming :)
My variant is better IMHO, because we don't use Windows and I even don't know if the app works on it.

To avoid "McCabe complexity" you can make 2 helper functions like SetupVAOUnixES2/SetupVAOWindows.

And as mentioned above, ES3 is equal for all Unix platforms.

Reply:
> My variant is better IMHO, because we don't use Windows and I even don't know if the app works on it.

I am not sure, that I see the connection between those. Regardless of that I have pushed your preferred variant. Please check.

Reply:
Please do not throw away Windows, at least leave it as is. I'm planning to resurrect it.

Reply:
Windows remains unchanged

Reply:
OK, the Windows specific part has been left as is.